node/node_syncing.go

```diff
@@ -208,8 +204,8 @@ func (node *Node) doBeaconSyncing() {
 					)
 					if err != nil {
 						node.beaconSync.AddLastMileBlock(beaconBlock)
-					} else if node.Consensus.IsLeader() {
-						// Only leader broadcast crosslink to avoid spamming p2p
+					} else if node.Consensus.IsLeader() || rand.Intn(100) <= 1 {
+						// Only leader or 2% of validators broadcast crosslink to avoid spamming p2p
```

Comment:
Leader is still broadcasting crosslink crosslink here. Why don't we need this logic in stream sync (a.k.a. insertHook)?

Reply:
Because this is for the live beacon sync. Meaning when there is a new beacon block produced, it's broadcasted live to the other shards and here is when that block is received. This is the time we should broadcast back the crosslink based on the new beacon block.

But for normal catch up sync. it can be from block 0 to 1M. and we shouldn't do the crosslink broadcast at this situation.

Reply:
But the insertHook is called for live beacon sync. It only deals with block received from pub-sub message. It is not called when syncing from block 0 to 1m.

Reply:
ok, So the broadcast here is for legacy sync. and the hooked broadcast is for stream sync. And they are doing exactly the same thing?

Reply:
Yep, exactly.

Reply:
ok, let me revert the hook change.

Reply:
reverted.

Reply:
Thanks